images/scripts/fedora.setup

```diff
@@ -175,11 +175,19 @@ fi
 dnf $DNF_OPTS -y upgrade
 dnf $DNF_OPTS -y install $ALL_DEPS
 
+# create/run this as a script, so that it can be re-run at image-prepare time with different repos
+cat > /var/lib/download-package-sets.sh <<EOF
+#!/bin/sh
+set -eu
+rm -rf /var/lib/download-package-sets
 mkdir -p /var/lib/package-sets/clevis
-dnf $DNF_OPTS download --destdir=/var/lib/package-sets/clevis $PACKAGE_SET_CLEVIS
+dnf \${DNF_OPTS:=} download --destdir=/var/lib/package-sets/clevis $(echo "$PACKAGE_SET_CLEVIS" | xargs)
```

Comment:
Hmm, what does the escaping do here? Don't we want to insert the options into the script? `DNF_OPTS` will not be defined when the script is run, no?

Reply:
I misunderstood this. I thought it was a knob for using this from "outside", but indeed once upon a time fedora.setup actually defined `$DNF_OPTS`. It doesn't any more, and the script is not `set -u`. I fixed that, too.